lib/neovim-plugin.nix

```diff
@@ -152,13 +159,62 @@
                     }
                   ))
                 ]
-                ++ (lib.optionals (!hasConfigAttrs) [
+                ++ lib.optionals (!hasConfigAttrs) [
                   (lib.optionalAttrs callSetup (setLuaConfig setupCode))
-                ])
-                ++ (lib.optionals hasConfigAttrs [
+                ]
```

Comment:
How should the `!hasConfigAttrs` case be handled when lazyloading?

Reply:
Looks like we only have one instance of no config attrs, in rustaceanvim, because they use global options for configuration. Seems like something you don't get to really lazy load through a manager and is handled by the plugin itself?

Reply:
Certainly for lz-n, there's nothing to lazyload if there's no setup function. lazy.nvim is a bit fancier though.

If we _did_ go down that route, maybe the `plugins.*.lazyload` option should only exist when `hasSetup` is true? Although I'm happy for that to be tackled in future work if it is too much for this PR.